### PR TITLE
Domains: Update Domain Management contact information section

### PR DESCRIPTION
**Pull request by rafaelgallani**
#### Changes proposed in this Pull Request
This PR creates the new redesigned contact information component, shown in the domain settings page for registered and connected domains. This is part of the domain management pages redesign project detailed in pcYYhz-m2-p2.

#### Preview
image
image
image

#### Testing instructions
- Build this branch locally or open the live Calypso link
- If you're in the live Calypso link, please append `?flags=domains/settings-page-redesign` to your URL to enable the appropriate feature flag
- Go to "Upgrades > Domains"
- Select one of your registered or mapped domains
- Ensure that:
  - The contact information card in the domain settings page looks like the screenshot above;
  - The privacy protection toggle works like in the old contact information page;
  - The privacy check (whether privacy is enabled or not) works the same as in the old contact information page;
  - The non-owner notice in the screenshot above is displayed when you're not the domain owner;
  - The correct information is displayed;
  - The links work correctly;
  - The "Edit" button takes you to the edit contact information page;
  - The "Manage consent" button takes you to the manage consent for personal data use page;
  - Also check that the mobile view looks correct.
  
  In short, make sure it matches the redesigned version in hlVh2q24ad6MCwlwNnE9PQ-fi-600%3A54818.

### CODE REVIEW

File: client/my-sites/domains/domain-management/contacts-privacy/contacts-card.jsx [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+			this.props.discloseDomainContactInfo( selectedSite.ID, name );
+		}
+	};
+
+	getPrivacyProtection() {
+		const { privateDomain, privacyAvailable } = this.props;
+		const { translate, isUpdatingPrivacy } = this.props;
+
+		const label = privateDomain
+			? translate( 'Privacy protection on' )
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/contacts-privacy/contacts-card.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'Privacy protection: on' )` *ES Score: 10*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-management-contact-information)

File: client/my-sites/domains/domain-management/contacts-privacy/contacts-card.jsx [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+			: translate( 'Privacy protection off' );
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/contacts-privacy/contacts-card.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **21** times:
`translate( 'Privacy protection: off' )` *ES Score: 11*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-management-contact-information)

File: client/my-sites/domains/domain-management/settings/index.tsx [hunk 3]
```diff
@@ -41,6 +45,35 @@ const Settings = ( props: SettingsPageProps ): JSX.Element => {
...
+		const contactInformation = findRegistrantWhois( whoisData );
+
+		if ( ! contactInformation ) {
+			props.requestWhois( props.selectedDomainName );
+			return null;
+		}
+
+		const contactInfoFullName = `${ contactInformation.fname } ${ contactInformation.lname }`;
+		const privacyProtectionLabel = privateDomain
+			? translate( 'privacy protection on' )
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/settings/index.tsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'Privacy protection: on' )` *ES Score: 10*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-management-contact-information)

File: client/my-sites/domains/domain-management/settings/index.tsx [hunk 4]
```diff
@@ -41,6 +45,35 @@ const Settings = ( props: SettingsPageProps ): JSX.Element => {
...
+			: translate( 'privacy protection off' );
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/settings/index.tsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **21** times:
`translate( 'Privacy protection: off' )` *ES Score: 11*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-management-contact-information)

File: client/my-sites/domains/domain-management/contacts-privacy/contacts-card.tsx [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+			props.discloseDomainContactInfo( selectedSite.ID, name );
+		}
+	};
+
+	const getPrivacyProtection = () => {
+		const { privateDomain, privacyAvailable } = props;
+		const { isUpdatingPrivacy } = props;
+
+		const label = privateDomain
+			? translate( 'Privacy protection on' )
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/contacts-privacy/contacts-card.tsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'Privacy protection: on' )` *ES Score: 10*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-management-contact-information)

File: client/my-sites/domains/domain-management/contacts-privacy/contacts-card.tsx [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+			: translate( 'Privacy protection off' );
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/contacts-privacy/contacts-card.tsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **21** times:
`translate( 'Privacy protection: off' )` *ES Score: 11*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-management-contact-information)

File: client/my-sites/domains/domain-management/settings/index.tsx [hunk 7]
```diff
@@ -56,6 +61,53 @@ const Settings = ( props: SettingsPageProps ): JSX.Element => {
 		</>
 	);
 
+	const getContactInformationAccordion = () => {
```
Comment (leonardost) [inline client/my-sites/domains/domain-management/settings/index.tsx]: I think there might be a problem here when viewing domains of which we are not owners - the `requestWhois` query will return an error, and the component's loading state will never be resolved. A quick and dirty fix would be adding a condition at the start of this method to check if the current user can manage this domain, something like

```
if ( ! domain.currentUserCanManage ) {
	return (
		<Accordion title="Contact information">
			<ContactsPrivacyInfo
				domains={ props.domains! }
				selectedSite={ props.selectedSite }
				selectedDomainName={ props.selectedDomainName }
			></ContactsPrivacyInfo>
		</Accordion>
	);
}
```

This duplicates some code, but I couldn't think of anything more elegant at the moment 😅

**Review by leonardost — CHANGES_REQUESTED** [review]
This is looking great, nice job! 👍  I tested with some different domains and found only one issue when viewing a domain where I'm not the owner - the loading state never resolves because the `whois` query returns an error. I've left a fix suggestion in the code. Other than that, works great! 😄

**Comment by rafaelgallani** [comment]
Changing base as the folder structure/methods usage in #59334 is more adequate - also to avoid a future conflict resolution when rebasing 😄

File: client/my-sites/domains/domain-management/settings/cards/contact-information/contacts-card.tsx [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+			props.discloseDomainContactInfo( selectedSite.ID, name );
+		}
+	};
+
+	const getPrivacyProtection = () => {
+		const { privateDomain, privacyAvailable } = props;
+		const { isUpdatingPrivacy } = props;
+
+		const label = privateDomain
+			? translate( 'Privacy protection on' )
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/settings/cards/contact-information/contacts-card.tsx:48]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'Privacy protection: on' )` *ES Score: 10*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-management-contact-information)

File: client/my-sites/domains/domain-management/settings/cards/contact-information/contacts-card.tsx [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+			: translate( 'Privacy protection off' );
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/settings/cards/contact-information/contacts-card.tsx:49]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **21** times:
`translate( 'Privacy protection: off' )` *ES Score: 11*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-management-contact-information)

File: client/my-sites/domains/domain-management/settings/index.tsx [hunk 10]
```diff
@@ -129,12 +136,71 @@ const Settings = ( {
...
+				selectedSite={ selectedSite }
+				selectedDomainName={ selectedDomainName }
+			></ContactsPrivacyInfo>
+		);
+
+		const contactInformation = findRegistrantWhois( whoisData );
+
+		const { privateDomain } = domain;
+		const privacyProtectionLabel = privateDomain
+			? translate( 'Privacy protection on', { textOnly: true } )
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/settings/index.tsx:165]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'Privacy protection: on' )` *ES Score: 10*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-management-contact-information)

File: client/my-sites/domains/domain-management/settings/index.tsx [hunk 11]
```diff
@@ -129,12 +136,71 @@ const Settings = ( {
...
+			: translate( 'Privacy protection off', { textOnly: true } );
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/settings/index.tsx:166]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **21** times:
`translate( 'Privacy protection: off' )` *ES Score: 11*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-management-contact-information)

File: client/my-sites/domains/domain-management/settings/cards/contact-information/contacts-card.tsx [hunk 12]
```diff
@@ -0,0 +1,188 @@
...
+
+		let privacyProtectionNote;
+		if ( ! privacyAvailable ) {
+			privacyProtectionNote = (
+				<p className="contact-information__toggle-item">
+					{ translate(
+						'Privacy protection is not available due to the registry’s policies. {{a}}Learn more{{/a}}',
+						{
+							components: {
+								a: <a href={ PRIVACY_PROTECTION } />,
```
Comment (gius80) [inline client/my-sites/domains/domain-management/settings/cards/contact-information/contacts-card.tsx]: I noticed that previous component had` target="blank"` for this link, maybe we can maintain the same approach here.

```suggestion
								a: <a href={ PRIVACY_PROTECTION } target="_blank" rel="noopener noreferrer" />,
```

File: client/my-sites/domains/domain-management/settings/cards/contact-information/contacts-card.tsx [hunk 13]
```diff
@@ -0,0 +1,188 @@
...
+		);
+	};
+
+	const getPrivacyProtectionRecommendationText = () => {
+		const { privacyAvailable } = props;
+		return privacyAvailable ? (
+			<p className="contact-information__toggle-item">
+				{ translate( 'We recommend keeping privacy protection on. {{a}}Learn more{{/a}}', {
+					components: {
+						a: <a href={ PUBLIC_VS_PRIVATE } />,
```
Comment (gius80) [inline client/my-sites/domains/domain-management/settings/cards/contact-information/contacts-card.tsx]: Same as https://github.com/Automattic/wp-calypso/pull/59194/files#r772218992

**Comment by gius80** [comment]
I re-checked, now everything looks ok! Good job! 🤙

I Left just a couple of nitpicky comments.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7052628

Thank you @rafaelgalani for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7052628).
